### PR TITLE
Add support for v flag to `regexp/negation` rule

### PR DESCRIPTION
**Pull request by ota-meshi**
related to #545

### CODE REVIEW

**Comment by RunDevelopment** [comment]
Relevant: https://bugs.chromium.org/p/chromium/issues/detail?id=1474268

**Comment by ota-meshi** [comment]
Thank you for sharing the link, and making some fixes!
It looks like the Chrome bug you shared hasn't been fixed yet, but I think that ready to review this rule.

**Comment by RunDevelopment** [comment]
The fix for the bug has been merged yesterday and will be part of Chrome 118. 118 will only be released starting on Oct 10 though, so it will take some time until we see the fix. But that's fine, we can release the rule.

File: lib/rules/negation.ts [hunk 1]
```diff
@@ -53,13 +63,12 @@ export default createRule("negation", {
                         // All other character sets are either case-invariant
                         // (/./, /\s/, /\d/) or inconsistent (/\w/).
 
-                        // FIXME: TS Error
-                        // @ts-expect-error -- FIXME
-                        const ccSet = toCharSet(ccNode, flags)
+                        // since we know that the property doesn't contain strings,
+                        // we can just get the characters of the character class
+                        // without worrying about the strings
+                        const ccSet = toUnicodeSet(ccNode, flags).chars
```
Comment (RunDevelopment) [inline lib/rules/negation.ts:69]: I would like to get rid of the `.chars` here. The reasoning explained in the comment isn't trivial, so I would like to simplify it to the code I suggested.

Unfortunately, we can't compare a `UnicodeSet` to a `CharSet` right now, but I'm currently working on that. I'm currently working on a patch for refa and RAA which will include some fixes + more ergonomics for `UnicodeSet`s. So my suggested code will work then.

```suggestion
                        const ccSet = toUnicodeSet(ccNode, flags)
```

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**